File: __Viejo/Formularios/Agregar/CS/add_condicion_venta.vb
Public Class add_condicion_venta
    Private Sub cmd_ok_Click(sender As Object, e As EventArgs) Handles cmd_ok.Click
        If txt_condicion.Text = "" Then
            MsgBox("El campo 'Condicion' es obligatorio y está vacio")
            Exit Sub
        ElseIf txt_vencimiento.Text = "" Then
            MsgBox("El campo 'Vencimiento' es obligatorio y está vacio")
            Exit Sub
        ElseIf txt_recargo.Text = "" Then
            MsgBox("El campo 'Recargo' es obligatorio y está vacio")
            Exit Sub
        End If

        Dim tmp As New condicion_venta

        tmp.condicion = txt_condicion.Text
        tmp.vencimiento = txt_vencimiento.Text
        tmp.recargo = txt_recargo.Text
        tmp.activo = chk_activo.Checked

        If edicion = True Then
            tmp.id_condicion_venta = edita_condicion_venta.id_condicion_venta
            If updateCondicion_venta(tmp) = False Then
                MsgBox("Hubo un problema al actualizar la condicion, consulte con su programdor", vbExclamation)
                closeandupdate(Me)
            End If
        Else
            addCondicion_venta(tmp)
        End If

        If chk_secuencia.Checked = True Then
            txt_condicion.Text = ""
            txt_vencimiento.Text = ""
            txt_recargo.Text = ""
            chk_activo.Checked = True
        Else
            closeandupdate(Me)
        End If
    End Sub

    Private Sub cmd_exit_Click(sender As Object, e As EventArgs) Handles cmd_exit.Click
        closeandupdate(Me)
    End Sub

    Private Sub Add_marcai_FormClosed(sender As Object, e As FormClosedEventArgs) Handles Me.FormClosed
        closeandupdate(Me)
    End Sub

    Private Sub Add_condicion_Load(sender As Object, e As EventArgs) Handles MyBase.Load

        chk_activo.Checked = True
        If edicion = True Or borrado = True Then
            chk_secuencia.Enabled = False
            txt_condicion.Text = edita_condicion_venta.condicion
            txt_vencimiento.Text = edita_condicion_venta.vencimiento
            txt_recargo.Text = edita_condicion_venta.recargo
            chk_activo.Checked = edita_condicion_venta.activo
        End If

        If borrado = True Then
            txt_condicion.Enabled = False
            txt_vencimiento.Enabled = False
            txt_recargo.Enabled = False
            chk_activo.Enabled = False
            cmd_ok.Visible = False
            cmd_exit.Visible = False
            'rb_aplicaVentas.Enabled = False
            'rb_aplicaCompras.Enabled = False
            Me.Show()
            If MsgBox("¿Está seguro que desea borrar esta condición?", vbYesNo + vbQuestion) = MsgBoxResult.Yes Then
                If (borrarCondicion_venta(edita_condicion_venta)) = False Then
                    If (MsgBox("Ocurrió un error al realizar el borrado de la condición, ¿desea intectar desactivarlo para que no aparezca en la búsqueda?" _
                     , MsgBoxStyle.Question + MsgBoxStyle.YesNo)) = vbYes Then
                        'Realizo un borrado lógico
                        If updateCondicion_venta(edita_condicion_venta, True) = True Then
                            MsgBox("Se ha podido realizar un borrado lógico, pero la condición no se borró definitivamente." + Chr(13) +
                                "Esto posiblemente se deba a que la condición, tiene operaciones realizadas y por lo tanto no podrá borrarse", vbInformation)
                        Else
                            MsgBox("No se ha podido borrar la condición, consulte con el programador")
                        End If
                    End If
                End If
            End If
            closeandupdate(Me)
        End If
    End Sub
End Class

File: Formularios/Agregar/add_condicion_venta.cs
using System;
using System.Windows.Forms;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;
using Centrex.Models;

namespace Centrex
{
    public partial class add_condicion_venta
    {
        public add_condicion_venta()
        {
            InitializeComponent();
        }

        private void cmd_ok_Click(object sender, EventArgs e)
        {
            if (string.IsNullOrEmpty(txt_condicion.Text))
            {
                Interaction.MsgBox("El campo 'Condicion' es obligatorio y está vacio");
                return;
            }
            else if (string.IsNullOrEmpty(txt_vencimiento.Text))
            {
                Interaction.MsgBox("El campo 'Vencimiento' es obligatorio y está vacio");
                return;
            }
            else if (string.IsNullOrEmpty(txt_recargo.Text))
            {
                Interaction.MsgBox("El campo 'Recargo' es obligatorio y está vacio");
                return;
            }

            var tmp = new CondicionVentaEntity();

            tmp.Condicion = txt_condicion.Text;
            tmp.Vencimiento = Conversions.ToInteger(txt_vencimiento.Text);
            tmp.Recargo = Conversions.ToDecimal(txt_recargo.Text);
            tmp.Activo = chk_activo.Checked;

            if (VariablesGlobales.edicion == true)
            {
                tmp.IdCondicionVenta = VariablesGlobales.edita_condicion_venta.IdCondicionVenta;
                if (condiciones_venta.updateCondicion_venta(tmp) == false)
                {
                    Interaction.MsgBox("Hubo un problema al actualizar la condición, consulte con su programador", Constants.vbExclamation);
                    closeandupdate(this);
                }
            }
            else
            {
                condiciones_venta.addCondicion_venta(tmp);
            }

            if (chk_secuencia.Checked == true)
            {
                txt_condicion.Text = "";
                txt_vencimiento.Text = "";
                txt_recargo.Text = "";
                chk_activo.Checked = true;
            }
            else
            {
                closeandupdate(this);
            }
        }

        private void cmd_exit_Click(object sender, EventArgs e)
        {
            closeandupdate(this);
        }

        private void Add_marcai_FormClosed(object sender, FormClosedEventArgs e)
        {
            closeandupdate(this);
        }

        private void Add_condicion_Load(object sender, EventArgs e)
        {
            chk_activo.Checked = true;
            if (VariablesGlobales.edicion == true | VariablesGlobales.borrado == true)
            {
                chk_secuencia.Enabled = false;
                txt_condicion.Text = VariablesGlobales.edita_condicion_venta.Condicion;
                txt_vencimiento.Text = VariablesGlobales.edita_condicion_venta.Vencimiento.ToString();
                txt_recargo.Text = VariablesGlobales.edita_condicion_venta.Recargo.ToString();
                chk_activo.Checked = VariablesGlobales.edita_condicion_venta.Activo;
            }

            if (VariablesGlobales.borrado == true)
            {
                txt_condicion.Enabled = false;
                txt_vencimiento.Enabled = false;
                txt_recargo.Enabled = false;
                chk_activo.Enabled = false;
                cmd_ok.Visible = false;
                cmd_exit.Visible = false;
                // rb_aplicaVentas.Enabled = False
                // rb_aplicaCompras.Enabled = False
                Show();
                if (Interaction.MsgBox("¿Está seguro que desea borrar esta condición?", (MsgBoxStyle)((int)Constants.vbYesNo + (int)Constants.vbQuestion)) == MsgBoxResult.Yes)
                {
                    if (condiciones_venta.borrarCondicion_venta(VariablesGlobales.edita_condicion_venta) == false)
                    {
                        if (Interaction.MsgBox("Ocurrió un error al realizar el borrado de la condición, ¿desea intentar desactivarla para que no aparezca en la búsqueda?", (MsgBoxStyle)((int)MsgBoxStyle.Question + (int)MsgBoxStyle.YesNo)) == Constants.vbYes)
                        {
                            // Realizo un borrado lógico
                            if (condiciones_venta.updateCondicion_venta(VariablesGlobales.edita_condicion_venta, true) == true)
                            {
                                Interaction.MsgBox("Se ha podido realizar un borrado lógico, pero la condición no se borró definitivamente." + "\r" + "Esto posiblemente se deba a que la condición tiene operaciones realizadas y por lo tanto no podrá borrarse", Constants.vbInformation);
                            }
                            else
                            {
                                Interaction.MsgBox("No se ha podido borrar la condición, consulte con el programador");
                            }
                        }
                    }
                }
                closeandupdate(this);
            }
        }
    }
}

